include/circt/Dialect/Calyx/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+add_circt_dialect(Calyx calyx)
+add_circt_doc(Calyx  -gen-dialect-doc Calyx Dialects/)
+
+set(LLVM_TARGET_DEFINITIONS Calyx.td)
+
+mlir_tablegen(CalyxEnums.h.inc -gen-enum-decls)
+mlir_tablegen(CalyxEnums.cpp.inc -gen-enum-defs)
+mlir_tablegen(CalyxAttrs.h.inc -gen-struct-attr-decls)
+mlir_tablegen(CalyxAttrs.cpp.inc -gen-struct-attr-defs)
+
+
+add_public_tablegen_target(MLIRCalyxEnumsIncGen)
```

Comment:
I think you'll also want `add_dependencies(circt-headers MLIRCalyxEnumsIncGen)`.

Reply:
The SV dialect CMakeLists might serve as a good example of the boilerplate here: https://github.com/llvm/circt/blob/main/include/circt/Dialect/SV/CMakeLists.txt

Reply:
Thanks!